File: packages/ios/Sources/Components/DynamicIsland/DynamicIsland.figma.swift
import SwiftUI
import Figma

struct KozmosDynamicIslandConnect: FigmaConnect {
    let component = KozmosDynamicIsland<EmptyView, EmptyView, EmptyView, EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8358"

    @FigmaEnum(
        "State",
        mapping: [
            "Compact": KozmosDynamicIsland<EmptyView, EmptyView, EmptyView, EmptyView>.IslandState.compact,
            "Expanded": KozmosDynamicIsland<EmptyView, EmptyView, EmptyView, EmptyView>.IslandState.expanded,
            "Minimal": KozmosDynamicIsland<EmptyView, EmptyView, EmptyView, EmptyView>.IslandState.minimal
        ]
    )
    var state: KozmosDynamicIsland<EmptyView, EmptyView, EmptyView, EmptyView>.IslandState = .compact

    var body: some View {
        KozmosDynamicIsland(
            state: self.state,
            expandedContent: { EmptyView() },
            compactLeading: { EmptyView() },
            compactTrailing: { EmptyView() },
            minimalContent: { EmptyView() }
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/DynamicIsland/DynamicIsland.figma.kt
package com.kozmos.components.dynamicisland

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8358")
class KozmosDynamicIslandConnect {
    @FigmaProperty(FigmaType.Enum, "State")
    val state: KozmosDynamicIslandState = Figma.mapping(
        "Compact" to KozmosDynamicIslandState.Compact,
        "Expanded" to KozmosDynamicIslandState.Expanded,
        "Minimal" to KozmosDynamicIslandState.Minimal
    )

    @Composable
    fun ComponentExample() {
        KozmosDynamicIsland(state = state)
    }
}
